### DISCUSSION TITLE
Add properties to Many-to-Many relationship

### DISCUSSION THREAD

**Discussion by dinhdangkhoa** [atrocore/atrocore#183] [Q&A] [answered]
Hello,
I'm currently in the modeling phase to store our data into AtroPIM.

We have "Color" and "Product". Each product can access to differents colors, and the price is depended on both product and color. I've created the table Color and the relationship Many-to-Many betwen these 2 tables.
Can I add a custom property "Price" into the middle table ? And is it possible to specify the price in the GUI when I associate colors to a product ?

Thank you.

**Comment by atrolex** [comment, marked as answer]
We have a premium module for that, called "configurable products". You can create a configurable product (which is your product template) and then create for this product different variants, with "color" set as configurable attribute. So you will be able to create different products, which will inherit all the product attributes from the parent and have own value for the "color" attribute. Each variant is a separate record, so it can have own prices and so on.

_If you still want to have your way:_
Adding properties on relationship is planned, but I cannot tell you for now when we are so far. It is a complicated task to make it configurable, but it can be achieved pretty simple via programming.

**Reply by dinhdangkhoa** [reply]
Thank you for your quick reply. I understand that it's complicated to make it generic. I think I will do it via programming. Have you any documentation for developers ? I'm searching around but I cannot find it. 
Thank you again.

**Reply by atrolex** [reply]
We plan to create it, but currently we have no resources for that. You can have a look at Developer Documentation for EspoCRM, we have a lot of similarities, because AtroPIM is based on EspoCRM.  But, still, there are already a lot of differences.

**Reply by dinhdangkhoa** [reply]
Thank you. I'll look into it.
